lib/skate/settings/user.ex

```diff
@@ -14,13 +13,18 @@ defmodule Skate.Settings.User do
     )
   end
 
+  def user_ids_for_route_ids(route_ids) do
+    Skate.Repo.all(from(u in users_for_route_ids_query(route_ids), select: u.id))
+  end
+
   def usernames_for_route_ids(route_ids) do
-    Skate.Repo.all(
-      from(rs in DbRouteSettings,
-        join: u in assoc(rs, :user),
-        where: fragment("?::varchar[] && ?", ^route_ids, rs.selected_route_ids),
-        select: u.username
-      )
+    Skate.Repo.all(from(u in users_for_route_ids_query(route_ids), select: u.username))
+  end
+
+  defp users_for_route_ids_query(route_ids) do
+    from(u in DbUser,
+      join: rs in assoc(u, :route_settings),
+      where: fragment("?::varchar[] && ?", ^route_ids, rs.selected_route_ids)
```

Comment:
Hmm, I'm having sort of a hard time parsing out exactly what this fragment does, could you explain it? I'm guessing it checks to see if the route IDs are contained in `selected_route_ids` somehow?

Reply:
The `&&` is a Boolean operator on arrays that returns true if the arrays have at least one element in common (https://www.postgresql.org/docs/12/functions-array.html). So this fragment gives us DB records that have at least one of the selected routes.

Reply:
Ah, okay. I was just totally unfamiliar with the Postgres array syntax I guess. Thanks!